lib/vector-core/buffers/src/disk_v2/ledger.rs

```diff
@@ -586,10 +604,57 @@ impl Ledger {
             last_flush: AtomicCell::new(Instant::now()),
             usage_handle,
         };
-        ledger.synchronize_buffer_usage();
+        ledger.update_buffer_size().await?;
 
         Ok(ledger)
     }
+
+    async fn update_buffer_size(&mut self) -> Result<(), LedgerLoadCreateError> {
```

Comment:
What happens with this approach if someone starts two vector processes with the same config at the same time?

Reply:
This is a good question. Maybe we should have a lockfile?

Reply:
We attempt to acquire an advisory lock when loading a buffer.  If one process has already acquired it, the other process would return an error and Vector should just end up shutting down.  We acquire the advisor lock before we do anything else, so it protects all of the initialization logic.

It's only advisory, so, it doesn't stop processes from being able to forcefully open and modify our buffer files, but, that's out of scope anyways.

Reply:
Cool.